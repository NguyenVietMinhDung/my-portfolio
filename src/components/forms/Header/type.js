import type { Props as SocialNetworks } from '../../commons/SocialNetworks/type';

export type Props = {
  socialNetworks: SocialNetworks,
  name: string,
  description: string,
  avatarUrl: string,
  btnTxt: string,
};
