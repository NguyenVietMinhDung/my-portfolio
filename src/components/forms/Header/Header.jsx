// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Header.scss';
import type { Props } from './type';
import { Button, SocialNetworks } from '../../commons';

const handleMouseMoveEvent = (event) => {
  const element = document.getElementById('dynamic-background');
  if (element) {
    const { clientX, clientY } = event;
    const { style } = element;
    style.setProperty('transform', `translate3d(${clientX * 0.04}px, ${clientY * 0.08}px, 0)`);
  }
};

const handleMouseOutEvent = () => {
  const element = document.getElementById('dynamic-background');
  if (element) {
    const { style } = element;
    style.setProperty('transform', 'translate3d(0, 0, 0)');
  }
};

const Header = (props: Props) => {
  const {
    name, description, avatarUrl, btnTxt, socialNetworks,
  } = props;
  return (
    <div
      id="header"
      styleName="header"
      onMouseMove={handleMouseMoveEvent}
      onMouseOut={handleMouseOutEvent}
    >
      <div
        id="dynamic-background"
        styleName="dynamic-background"
      />
      <div styleName="content">
        <div styleName="introduction">
          <div styleName="greeting">HELLO, I&apos;M</div>
          <div styleName="name">{name}</div>
          <div styleName="description">{description}</div>
          <Button
            text={btnTxt}
            handleClickEvent={() => null}
          />
        </div>
        <div styleName="avatar">
          <img
            src={avatarUrl}
            alt=""
          />
        </div>
      </div>
      <div styleName="social-networks">
        <SocialNetworks items={socialNetworks} />
      </div>
    </div>
  );
};

export default memo<Props>(CSSModules(Header, styles));
