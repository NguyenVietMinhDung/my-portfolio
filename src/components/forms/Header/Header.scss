@import '../../../styles/abstracts/variables';
@import '../../../styles/abstracts/mixins';

.header {
	position: relative;

	display: flex;
	overflow: hidden;
	align-items: center;
	flex-flow: column wrap;

	.dynamic-background {
		position: absolute;
		top: 0;
		left: -100px;

		width: calc(100% + 200px);
		height: 100%;

		transition: all 0.25s ease-out 0s;

		background: url($header-bg-img) no-repeat center;
		background-size: cover;
	}

	.content {
		z-index: 1;

		display: flex;
		align-items: center;
		flex-flow: row wrap;

		.introduction {
			flex: 1;

			.name {
				@media (min-width: $layout-breakpoint) {
					@include font(inherit, uppercase, $black-color, 60px, 600);
				}

				@media (max-width: $layout-breakpoint) {
					@include font(inherit, uppercase, $black-color, 38px, 600);
				}
			}

			.greeting {
				@media (min-width: $layout-breakpoint) {
					@include font(inherit, uppercase, $violet-color, 30px, 600);
				}

				@media (max-width: $layout-breakpoint) {
					@include font(inherit, uppercase, $violet-color, 24px, 600);
				}
			}

			.description {
				padding-bottom: 30px;
			}
		}

		@media (min-width: $layout-breakpoint) {
			width: $layout-breakpoint;
			padding-top: 150px;

			.avatar {
				position: relative;

				display: flex;
				flex: 1;
				flex-flow: row wrap;
				justify-content: center;

				img {
					position: relative;

					width: 300px;
					height: 450px;
				}

				&::before {
					position: absolute;
					top: 60px;

					width: 265px;
					height: 350px;

					content: '';

					border: 20px solid $violet-color;
				}

				&::after {
					position: absolute;
					bottom: 0;

					width: 305px;
					height: 20px;

					content: '';

					background-color: $violet-color;
				}
			}
		}

		@media (max-width: $layout-breakpoint) {
			width: auto;
			padding-top: 200px;

			.avatar {
				display: none;
			}
		}
	}

	.social-networks {
		position: relative;

		display: flex;
		flex-flow: row wrap;
		justify-content: flex-start;

		&::before {
			position: absolute;
			top: 50%;
			left: -100%;

			width: 100%;
			height: 2px;

			content: '';
			transform: translateY(-50%);

			background-color: $violet-color;
		}

		@media (min-width: $layout-breakpoint) {
			width: $layout-breakpoint;
			margin-top: 0;
			padding: 0;
		}

		@media (max-width: $layout-breakpoint) {
			width: calc(100% - 30px);
			margin-top: 50px;
		}
	}
}
