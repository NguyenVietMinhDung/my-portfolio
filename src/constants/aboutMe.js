export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
