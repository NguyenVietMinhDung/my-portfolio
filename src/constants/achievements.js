export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
