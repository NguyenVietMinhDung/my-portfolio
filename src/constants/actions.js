export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
