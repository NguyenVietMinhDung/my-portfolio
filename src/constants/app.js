import facebookIconUrl from '../assets/images/facebook.png';
import instagramIconUrl from '../assets/images/instagram.png';
import linkedInIconUrl from '../assets/images/linkedin.png';

export const LOADER_SIZE = 50;
export const LOADER_SIZE_UNIT = 'px';
export const LOADER_COLOR = '#754ef9';
export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/dungvmn1995',
    iconUrl: linkedInIconUrl,
  },
];
export logoUrl from '../assets/images/logo.png';
