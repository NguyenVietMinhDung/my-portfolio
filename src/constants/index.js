import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
};
