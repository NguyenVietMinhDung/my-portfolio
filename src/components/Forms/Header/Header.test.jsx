import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import constants from '../../../constants';
import Header from './Header';

const {
  header: {
    NAME, DESCRIPTION, BUTTON_TEXT, avatarUrl,
  },
  app: { socialNetworks },
} = constants;

const defaultProps = {
  btnTxt: BUTTON_TEXT,
  name: NAME,
  description: DESCRIPTION,
  socialNetworks,
  avatarUrl,
};

describe('<Header />', () => {
  test('renders correctly', () => {
    const component = shallow(<Header {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('handle mousemove event', () => {
    const component = mount(<Header {...defaultProps} />);
    component.find('#header').simulate('mouseMove', { clientX: 100, clientY: 100 });
  });
  test('handle mouseout event', () => {
    const component = mount(<Header {...defaultProps} />);
    component.find('#header').simulate('mouseOut');
  });
});
