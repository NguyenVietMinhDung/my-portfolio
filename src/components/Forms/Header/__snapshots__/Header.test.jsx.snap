// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<div
  className="header"
  id="header"
  onMouseMove={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="dynamic-background"
    id="dynamic-background"
  />
  <div
    className="content"
  >
    <div
      className="introduction"
    >
      <div
        className="greeting"
      >
        HELLO, I'M
      </div>
      <div
        className="name"
      >
        Dung Nguyen
      </div>
      <div
        className="description"
      >
        A Freelance UI Designer & Web Developer
      </div>
      <Memo(Button)
        handleClickEvent={[Function]}
        text="DOWNLOAD MY CV"
      />
    </div>
    <div
      className="avatar"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="social-networks"
  >
    <Memo(SocialNetworks)
      items={
        Array [
          Object {
            "iconUrl": "test-file-stub",
            "name": "Facebook",
            "url": "https://www.facebook.com/minh.dung.391",
          },
          Object {
            "iconUrl": "test-file-stub",
            "name": "Instagram",
            "url": "https://www.instagram.com/dungvmnguyen",
          },
          Object {
            "iconUrl": "test-file-stub",
            "name": "LinkedIn",
            "url": "https://www.linkedin.com/in/dungvmn1995",
          },
        ]
      }
    />
  </div>
</div>
`;
